submissions/tavor118sn/task_friends_app/script.js

```diff
@@ -0,0 +1,153 @@
+  const cardHeader = document.createElement('h2');
+  cardHeader.classList.add('user-full-name');
+  cardHeader.innerHTML = `${userInfo.name.first} ${userInfo.name.last}`;
+  userCard.appendChild(cardHeader);
+
+  const genderElem = document.createElement('p');
+  genderElem.innerHTML = `Gender: <strong>${userInfo.gender}</strong>`;
+  userCard.appendChild(genderElem);
+
+  const ageElem = document.createElement('p');
+  ageElem.innerHTML = `Age: <strong>${userInfo.dob.age}</strong>`;
+  ageElem.classList.add('user-age');
+  userCard.appendChild(ageElem);
+
+  return userCard;
+};
+
+const generateFriendsListing = (friendList) => {
+  CARD_LIST.innerHTML = '';
+  const fragment = document.createDocumentFragment();
+
+  friendList.forEach((elem) => {
+    fragment.appendChild(createCard(elem));
+  });
+
+  CARD_LIST.appendChild(fragment);
+};
+
+// Load users
+
+async function loadUserList() {
+  const userFields = ['gender', 'name', 'email', 'picture', 'dob'];
+  const siteUrl = 'https://randomuser.me/api/';
+  const baseUrl = new URL(siteUrl);
+  baseUrl.searchParams.set('nat', 'us');
+  baseUrl.searchParams.set('results', '12');
+  baseUrl.searchParams.set('inc', userFields.join());
+
+  // try to make request at least 5 times in case of errors
+  for (let i = 0; i < 5; i++) {
+    try {
+      const response = await fetch(baseUrl);
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      } else {
+        const friends = await response.json();
+        return friends.results;
+      }
+    } catch (e) {
+      console.log(e);
+    }
+  }
+}
+
+// Sorting and filtering
+
+const getFormData = (form) => {
+  return {
+    userSort: form.userSort.value,
+    userFilterGender: form.userFilterGender.value,
+  };
+};
+
+const filterByGender = (gender) => {
+  let friendList = [...friends];
+  if (gender !== 'all') {
+    friendList = friendList.filter((user) => user.gender === gender);
+  }
+  return friendList;
+};
+
+const filterByName = (event) => {
+  const enteredFirstName = event.target.value;
+
+  let friendList = [...friends];
+  friendList = friendList.filter((user) =>
+    (user.name.first + user.name.last)
+      .toLowerCase()
+      .includes(enteredFirstName.toLowerCase())
+  );
+
+  generateFriendsListing(friendList);
+};
+
+const sortByLastNameAsc = (a, b) => {
+  const nameA = a.name.last.toUpperCase();
+  const nameB = b.name.last.toUpperCase();
+  return nameA < nameB ? -1 : nameA > nameB ? 1 : 0;
+};
+
+const sortByLastNameDesc = (a, b) => {
+  const nameA = a.name.last.toUpperCase();
+  const nameB = b.name.last.toUpperCase();
+  return nameA < nameB ? 1 : nameA > nameB ? -1 : 0;
+};
+
+const sortByAgeAsc = (a, b) => {
+  return a.dob.age - b.dob.age;
+};
+
+const sortByAgeDesc = (a, b) => {
+  return b.dob.age - a.dob.age;
+};
+
+const sortFuncMapper = {
+  sortByAgeAsc,
+  sortByAgeDesc,
+  sortByLastNameAsc,
+  sortByLastNameDesc,
+};
+
+const sortAndFilterListing = (event) => {
```

Comment:
![image](https://user-images.githubusercontent.com/2997359/105187171-2791d980-5b3b-11eb-85a0-3a21db46b948.png)

Please, use appropriate English words :)

Also, function names should be `verb`